### PR TITLE
Add includeOverQuota option to search

### PR DESCRIPTION
**Pull request by mbarton**
Filters out images that are over quota by inserting the equivalent of a `-supplier:XXX` chip for each supplier that is over quota.

### CODE REVIEW

File: media-api/app/MediaApiComponents.scala [hunk 1]
```diff
@@ -72,7 +72,7 @@ class MediaApiComponents(context: Context) extends GridComponents(context) {
 
   val imageResponse = new ImageResponse(config, s3Client, usageQuota)
 
-  val mediaApi = new MediaApi(auth, messageSender, elasticSearch, imageResponse, config, controllerComponents, s3Client, mediaApiMetrics)
+  val mediaApi = new MediaApi(auth, messageSender, elasticSearch, imageResponse, usageQuota.usageStore, config, controllerComponents, s3Client, mediaApiMetrics)
```
Comment (tonytw1) [inline media-api/app/MediaApiComponents.scala]: Would be nice to extract the usageStore to a top level component and then insert just that, but not to fussed about it.

File: media-api/app/controllers/MediaApi.scala [hunk 2]
```diff
@@ -286,16 +288,32 @@ class MediaApi(
...
+          // Annoyingly `NonEmptyList` and `IList` don't have `mkString`
+          errors.map(_.message).list.reduce(_ + ", " + _), List(searchLink))
+        ),
+        params => respondSuccess(params)
+      )
+    }
+  }
+
+  private def addOverQuotaFilters(searchParams: SearchParams): Future[SearchParams] = {
+    usageStore.getUsageStatus().map { result =>
```
Comment (tonytw1) [inline media-api/app/controllers/MediaApi.scala]: Variable name result is abit overloaded in this context; it's usageStatus?

**Comment by mbarton** [comment]
[Option to display pics that haven't exceeded quota](https://trello.com/c/rEPzHoyX/718-option-to-display-pics-that-havent-exceeded-quota)

**Comment by mbarton** [comment]
Closing until the work gets picked up again

**Pull request closed without merging**